tests/unit/moduleapi/datatype2.tcl

```diff
@@ -225,19 +214,21 @@ start_server {tags {"modules"}} {
         assert_equal $digest0 $digest1
     }
 
-    test "datatype2: test memusage" {
-        r flushall
+    if {[string match {*jemalloc*} [s mem_allocator]]} {
```

Comment:
in fact, we can ignore the indentation and make the diff smaller, like
also maybe we need a comment to explain why we do the skip

```
if {[string match {*jemalloc*} [s mem_allocator]]} {
    test "datatype2: test memusage" {
        r flushall

        set data k1
        assert_equal 3 [r mem.alloc k1 3]
        assert_equal [string length $data] [r mem.write k1 1 $data]
        assert_equal $data [r mem.read k1 1]

        set data k2
        assert_equal 3 [r mem.alloc k2 3]
        assert_equal [string length $data] [r mem.write k2 0 $data]
        assert_equal $data [r mem.read k2 0]

        assert_equal [r memory usage k1] [r memory usage k2] 
    }
}
```

Reply:
I don’t think lose formatting in order to get smaller diff is a good choice, otherwise I would rather give up other formatting parts.

Reply:
The other format is small and i think is needed...

This part, i think both way are ok, i personally just expect this more (i think oran is the same) :)

![image](https://user-images.githubusercontent.com/22811481/147456130-f7185b6c-dfb0-4b91-ab81-e1c45f18c633.png)

> This usually does not break the blame log, i think

Reply:
in the test suite, when we add an `if` or a `for` that indents a lot of code, in many cases i ask not to indent, and treat that `if` as an `#ifdef` (no indentation).
the advantages are:
1. less damage to blame log (easier to find the reason for some change)
2. less merge conflicts for old PRs, other forks of redis, and backport to older versions.
3. easier to review the actual changes since GH doesn't show all lines as new / deleted.

in this case it's not a major concern, so i don't mind either way.

however, correct me if i'm wrong, is there any reason to comment that whole test, and not just the assertion at the end?
i didn't pay close attention, but if we can keep running the majority of the code for better coverage, and just skip the last assertion, it'll be good.

even better, we can make a `assert_memory` utility method in utils.tcl, pass both values to it, and ignore them if not using jemalloc.
the advantage is that we still run the MEMORY USAGE command (coverage), and we can use this method in other tests in the future (ones that rely on exact memory usage results)

Reply:
Add `memory_usage`  function.